pkg/reconciler/cronjobsource/cronjobsource.go

```diff
@@ -201,40 +220,41 @@ func checkResourcesStatus(src *v1alpha1.CronJobSource) error {
 }
 
 func (r *Reconciler) createReceiveAdapter(ctx context.Context, src *v1alpha1.CronJobSource, sinkURI string) (*appsv1.Deployment, error) {
-
 	if err := checkResourcesStatus(src); err != nil {
 		return nil, err
 	}
 
-	ra, err := r.getReceiveAdapter(ctx, src)
-	if err != nil && !apierrors.IsNotFound(err) {
-		logging.FromContext(ctx).Error("Unable to get an existing receive adapter", zap.Error(err))
-		return nil, fmt.Errorf("getting receive adapter: %v", err)
-	}
 	adapterArgs := resources.ReceiveAdapterArgs{
 		Image:   r.env.Image,
 		Source:  src,
 		Labels:  resources.Labels(src.Name),
 		SinkURI: sinkURI,
 	}
 	expected := resources.MakeReceiveAdapter(&adapterArgs)
-	if ra != nil {
-		if podSpecChanged(ra.Spec.Template.Spec, expected.Spec.Template.Spec) {
-			ra.Spec.Template.Spec = expected.Spec.Template.Spec
-			if ra, err = r.KubeClientSet.AppsV1().Deployments(src.Namespace).Update(ra); err != nil {
-				return ra, fmt.Errorf("updating receive adapter: %v", err)
-			}
-			logging.FromContext(ctx).Info("Receive Adapter updated.", zap.Any("receiveAdapter", ra))
+
+	ra, err := r.KubeClientSet.AppsV1().Deployments(src.Namespace).Get(expected.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		ra, err = r.KubeClientSet.AppsV1().Deployments(src.Namespace).Create(expected)
+		msg := "Deployment created"
```

Comment:
no need for this msg... just use it in the Eventf call. Or use it in both places

Reply:
The reason why I added this is I couldn't find a way to get it to look nice otherwise. Initially I had:

```golang
r.Recorder.Eventf(src, corev1.EventTypeNormal, cronJobSourceDeploymentCreated, "Deployment created: error %v", err)
```

Which looked fine when there was an error, but looked bad when there wasn't (`Deployment created: error <nil>`). I like adding the error keyword, which is why I ended up with the current code. Overall, I want two distinct messages, "Deployment created" and "Deployment created, error: %v". Do you have a suggestion on how to accomplish that?

Reply:
sorry, my bad... Misread the code, they are indeed different logs...